src/test/scala/io/iohk/ethereum/network/rlpx/RLPxConnectionHandlerSpec.scala

```diff
@@ -70,6 +73,23 @@ class RLPxConnectionHandlerSpec extends FlatSpec with Matchers with MockFactory
     connection.expectNoMsg()
   }
 
+  it should "close the connection when Ack timeout happens" in new TestSetup {
+    (mockMessageCodec.encodeMessage _).expects(Ping(): MessageSerializable).returning(ByteString("ping encoded")).anyNumberOfTimes()
+
+    setupRLPxConnection()
+
+    rlpxConnectionParent watch rlpxConnection
+
+    rlpxConnection ! RLPxConnectionHandler.SendMessage(Ping())
+    connection.expectMsg(Tcp.Write(ByteString("ping encoded"), RLPxConnectionHandler.Ack))
+
+    //Wait for Ack timeout
+    Thread.sleep(rlpxConfiguration.waitForTcpAckTimeout.toMillis)
```

Comment:
@rtkaczyk Did you find a better way of doing this? Nico, It's not possible to use `eventually` here?